src/NLog.Targets.AppCenter/AppCenterTarget.cs

```diff
@@ -192,10 +199,65 @@ private void TrackEvent(string eventName, Exception exception, IDictionary<strin
                 properties = properties ?? new Dictionary<string, string>(1);
                 if (properties.Count < 20)
                     properties["EventName"] = eventName;
-                Microsoft.AppCenter.Crashes.Crashes.TrackError(exception, properties);
+
+                var path = RenderLogEvent(PathToCrashAttachmentDirectory, LogEventInfo.CreateNullEvent());
+                if (!string.IsNullOrEmpty(path))
+                {
+                    var errorAttachmentLogs = GetCompressedErrorAttachmentLogs(path);
+                    Microsoft.AppCenter.Crashes.Crashes.TrackError(exception, properties, errorAttachmentLogs.ToArray());
+                }
+                else
+                    Microsoft.AppCenter.Crashes.Crashes.TrackError(exception, properties);
             }
 
             Microsoft.AppCenter.Analytics.Analytics.TrackEvent(eventName, properties);
         }
+
+        /// <remarks>
+        ///     The maximum attached file (compressed) size is 1mb.
+        ///     The maximum attached file count is 10.
+        /// </remarks>
+        private List<Microsoft.AppCenter.Crashes.ErrorAttachmentLog> GetCompressedErrorAttachmentLogs(string path)
+        {
+            int attachedCount = 0;
+
+            var errorAttachements = new List<Microsoft.AppCenter.Crashes.ErrorAttachmentLog>();
+            var directoryToCompress = new System.IO.DirectoryInfo(path);

```

Comment:
Please add a try-catch around the entire logic, since File-operations are known to fail often (Ex. directory does not exist, directory is not accessible for user, files are open by other application and cannot be read, etc.)